conans/client/generators/msbuild.py

```diff
@@ -0,0 +1,198 @@
+        if os.path.isfile(multi_path):
+            content_multi = load(multi_path)
+        else:
+            content_multi = self.dep_props
+
+        # parse the multi_file and add a new import statement if needed
+        dom = minidom.parseString(content_multi)
+        import_group = dom.getElementsByTagName('ImportGroup')[0]
+        children = import_group.getElementsByTagName("Import")
+        for dep in deps:
+            conf_props_name = "conan_%s.props" % dep
+            for node in children:
+                if conf_props_name == node.getAttribute("Project"):
+                    # the import statement already exists
+                    break
+            else:
+                # create a new import statement
+                import_node = dom.createElement('Import')
+                import_node.setAttribute('Project', conf_props_name)
+                # add it to the import group
+                import_group.appendChild(import_node)
+        content_multi = dom.toprettyxml()
+        # To remove all extra blank lines
+        content_multi = "\n".join(line for line in content_multi.splitlines() if line.strip())
+        return content_multi
+
+    def _dep_conf(self, name, cpp_info):
+        def has_valid_ext(lib):
+            ext = os.path.splitext(lib)[1]
+            return ext in VALID_LIB_EXTENSIONS
+
+        t = "   <Import Project=\"{}\" Condition=\"'$(conan_{}_props_imported)' != 'True'\"/>"
+        transitive_imports = []
+        for dep_name in cpp_info.public_deps:
+            conf_props_name = "conan_%s.props" % dep_name
+            transitive_imports.append(t.format(conf_props_name, dep_name))
+        transitive_imports = os.linesep.join(transitive_imports)
+
+        fields = {
+            'name': name,
+            'transitive_imports': transitive_imports,
+            'bin_dirs': "".join("%s;" % p for p in cpp_info.bin_paths),
+            'res_dirs': "".join("%s;" % p for p in cpp_info.res_paths),
+            'include_dirs': "".join("%s;" % p for p in cpp_info.include_paths),
+            'lib_dirs': "".join("%s;" % p for p in cpp_info.lib_paths),
+            'libs': "".join(['%s.lib;' % lib if not has_valid_ext(lib)
+                             else '%s;' % lib for lib in cpp_info.libs]),
+            'system_libs': "".join(['%s.lib;' % sys_dep if not has_valid_ext(sys_dep)
+                                    else '%s;' % sys_dep for sys_dep in cpp_info.system_libs]),
+            'definitions': "".join("%s;" % d for d in cpp_info.defines),
+            'compiler_flags': " ".join(cpp_info.cxxflags + cpp_info.cflags),
+            'linker_flags': " ".join(cpp_info.sharedlinkflags),
+            'exe_flags': " ".join(cpp_info.exelinkflags)
+        }
+        formatted_template = self.dep_conf_props.format(**fields)
+        return formatted_template
+
+    @property
+    def content(self):
+        print("*** The 'msbuild' generator is EXPERIMENTAL ***")
```

Comment:
Probably it is better to send this output to `self.conanfile.output`

```suggestion
        self.conanfile.output.warn("*** The 'msbuild' generator is EXPERIMENTAL ***")
```

Reply:
This is exactly the assert that was failing with ``conan install <ref> -g msbuild``. The output is blocked with an assert for "virtual" nodes.

Reply:
I don't like the print either, but let leave this for later then